plugins/src/android/OptimoveSDKPlugin.java

```diff
@@ -9,8 +11,167 @@
 
 public class OptimoveSDKPlugin extends CordovaPlugin {
 
+    private static final String SET_USER_ID = "setUserId";
+    private static final String SET_USER_EMAIL = "setUserEmail";
+    private static final String REPORT_EVENT = "reportEvent";
+    private static final String REPORT_SCREEN_VISIT = "reportScreenVisit";
+    private static final String REGISTER_USER = "registerUser";
+    private static final String GET_VISITOR_ID = "getVisitorId";
+    private static final String GET_CURRENT_USER_IDENTIFIER = "getCurrentUserIdentifier";
+
     @Override
     public boolean execute(String action, JSONArray args, CallbackContext callbackContext) throws JSONException {
+        switch (action) {
+            case SET_USER_ID:
+                String userId = args.getString(0);
+                this.setUserId(userId, callbackContext);
+                return true;
+            case SET_USER_EMAIL:
+                String email = args.getString(0);
+                this.setUserEmail(email, callbackContext);
+                return true;
+            case REPORT_EVENT:
+                String eventName = args.getString(0);
+                String params = args.getString(1);
+                if (params == "null") {
```

Comment:
I think it should be possible to pass through an actual `null` or dictionary from JS.

See: https://github.com/Kumulos/KumulosSdkCordova/blob/81a605e4f98afe103dd6ab1e9b03b75df1aa7007/src/android/KumulosSDKPlugin.java#L201

(i.e. don't need to JSON encode it on the sender side)